### ISSUE TITLE
Question: Generating request- and response types for openapi 2.0?

### ISSUE THREAD

**Issue by tstelzer** [drwpow/openapi-typescript#437]
What is the stance on automatically generating request and response types? Names could be generated from `operationId`, or from the request path, e.g. 

```yaml
paths:
  get:
    thing:
```

Would become `GetThingRequest` and `GetThingResponse200` or something like that.

**Comment by gr2m** [comment]
Isn't that what `openapi-typescript` is already doing today? Not exactly in the format you ask, but as an example, these are the types generated from GItHub's OpenAPI spec, and types for both, operations and response are pretty easily to access:

https://github.com/octokit/openapi-types.ts/blob/main/generated/types.ts

What would you like to be changed, exactly?

**Comment by tstelzer** [comment]
Does it for openapi 2.0 though? 

Input:

```yaml
swagger: "2.0"
info:
  description: This is some API.
  version: 1.0.0
  title: My API
basePath: /api/v1
consumes:
  - application/json
  - multipart/form-data
produces:
  - application/json
schemes:
  - http
  - https
paths:
  /thing/{id}:
    get:
      summary: Get a Thing
      description: Get a Thing
      parameters:
        - in: path
          name: id
          required: true
          type: string
          description: The id of the thing.
      responses:
        200:
          description: The thing
          schema:
            $ref: "#/definitions/Thing"

definitions:
  Thing:
    type: object
    description: A thing
    properties:
      id:
        type: string
        description: Uniquely identifies this thing.
```

Output:

```ts
/**
 * This file was auto-generated by openapi-typescript.
 * Do not make direct changes to the file.
 */

export interface definitions {
  /**
   * A thing
   */
  Thing: {
    /**
     * Uniquely identifies this thing.
     */
    id?: string;
  };
}
```

**Comment by gr2m** [comment]
I can't help with v2.0 sorry, I'm only working with OpenAPI v3.0 myself

**Comment by tstelzer** [comment]
> I can't help with v2.0 sorry, I'm only working with OpenAPI v3.0 myself

I would love to say the same, but we're currently forced to work with OpenAPI 2, so I have no choice.

**Comment by drwpow** [comment]
> What is the stance on automatically generating request and response types?

Hey @tstelzer! Yes this is a frequently-requested feature addition for v2. This library started off only focusing on v2 `definitions` and nothing else, then we added v3 support which was a big effort, and only recently we started generating request/response for v3, but I don’t think it’s complete yet.

So to answer your question, the only reason we don’t do this for v2 yet is just time. Would love a PR if you’re able to provide one! Otherwise we _will_ add support for this eventually; it’s just tough to give an estimate on when. I haven’t had a lot of time to devote to this project nonstop lately.
